x-pack/plugins/enterprise_search/server/routes/enterprise_search/analytics.ts

```diff
@@ -157,4 +172,34 @@ export function registerAnalyticsRoutes({
       }
     })
   );
+
+  router.get(
+    {
+      path: '/internal/enterprise_search/analytics/events/{id}/exists',
+      validate: {
+        params: schema.object({
+          id: schema.string(),
+        }),
+      },
+    },
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const { client } = (await context.core).elasticsearch;
+
+      try {
+        const eventsIndexExists = await analyticsEventsIndexExists(client, request.params.id);
+
+        if (!eventsIndexExists) {
```

Comment:
I'm not sure. It's going to be not REST-compliant then, since we try to call `exists` in the scope of passed `{collectionId}`. If it was `events/exists/{collectionId}`, then it makes sense. Also, I wanted to be compliant with the Elasticsearch API, which returns 404 if the index doesn't exist and returns 200 with the payload when it exists